lib/vfs/src/host_fs.rs

```diff
@@ -0,0 +1,648 @@
+use crate::*;
+#[cfg(feature = "enable-serde")]
+use serde::{de, Deserialize, Serialize};
+use std::convert::TryInto;
+use std::fs;
+use std::io::{Read, Seek, Write};
+use std::path::{Path, PathBuf};
+use std::time::SystemTime;
+
+#[derive(Debug, Default, Clone)]
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+pub struct HostFileSystem;
+
+impl FileSystem for HostFileSystem {
+    fn read_dir(&self, path: &Path) -> Result<ReadDir, FsError> {
+        let read_dir = std::fs::read_dir(path)?;
+        let data = read_dir
+            .map(|entry| -> Result<DirEntry, _> {
+                let entry = entry?;
+                let metadata = entry.metadata()?;
+                Ok(DirEntry {
+                    path: entry.path(),
+                    metadata: Ok(fs_metadata_to_metadata(metadata)?),
+                })
+            })
+            .collect::<Result<Vec<DirEntry>, std::io::Error>>()
+            .map_err::<FsError, _>(Into::into)?;
+        Ok(ReadDir::new(data))
+    }
+    fn create_dir(&self, path: &Path) -> Result<(), FsError> {
+        fs::create_dir(path).map_err(Into::into)
+    }
+    fn remove_dir(&self, path: &Path) -> Result<(), FsError> {
+        fs::remove_dir(path).map_err(Into::into)
+    }
+    fn rename(&self, from: &Path, to: &Path) -> Result<(), FsError> {
+        fs::rename(from, to).map_err(Into::into)
+    }
+
+    fn remove_file(&self, path: &Path) -> Result<(), FsError> {
+        fs::remove_file(path).map_err(Into::into)
+    }
+    fn new_open_options(&self) -> OpenOptions {
+        OpenOptions::new(Box::new(HostFileOpener))
+    }
+
+    fn metadata(&self, path: &Path) -> Result<Metadata, FsError> {
+        fs::metadata(path)
+            .and_then(fs_metadata_to_metadata)
+            .map_err(Into::into)
+    }
+}
+
+fn fs_metadata_to_metadata(metadata: fs::Metadata) -> Result<Metadata, std::io::Error> {
+    use std::time::UNIX_EPOCH;
+    let filetype = metadata.file_type();
+    let (char_device, block_device, socket, fifo) = {
+        #[cfg(unix)]
+        {
+            use std::os::unix::fs::FileTypeExt;
+            (
+                filetype.is_char_device(),
+                filetype.is_block_device(),
+                filetype.is_socket(),
+                filetype.is_fifo(),
+            )
+        }
+        #[cfg(not(unix))]
```

Comment:
```suggestion
        #[cfg(windows)]
```

Reply:
See https://github.com/wasmerio/wasmer/pull/2491#discussion_r689570334. tl;dr it's fixed by https://github.com/wasmerio/wasmer/pull/2524.